src/lib/Liblog/pbs_log.c

```diff
@@ -332,6 +333,63 @@ log_init(void)
 #endif
 }
 
+/**
+ * @brief
+ *      Add interface information to log
+ *
+ * @par Side Effects:
+ *      None
+ *
+ * @par MT-safe: Yes
+ *
+ */
+
+void
+log_add_if_info()
+{
+	char tbuf[LOG_BUF_SIZE];
+	char inet_family[10], msg[LOG_BUF_SIZE], temp[LOG_BUF_SIZE];
+	int i;
+	struct net_if_info *ni, *curr;
+	
+	ni = (struct net_if_info*)malloc(sizeof(struct net_if_info));
+
+	memset(msg, '\0', sizeof(msg));
+	memset(temp, '\0', sizeof(temp));
+
+	get_if_info(ni, msg);
+
+	curr = ni;
+
+	if(strlen(msg)) { /* Adding error message to log */
+		strncpy(tbuf, msg, LOG_BUF_SIZE);
```

Comment:
The other way is to use LOG_BUF_SIZE - 1 in the strncpy or declare tbuf with size LOG_BUF_SIZE + 1, so that the size asked to copy is always less than the dest buffer size, resulting in a NULL termination always!